src/gtest/test_libzendoo.cpp

```diff
@@ -1973,4 +1982,140 @@ TEST(CctpLibrary, TestInvalidProofVkWhenOversized)
     EXPECT_FALSE(vkInvalid.IsValid());
 
     //TODO: Might be useful to test the same behaviour with bit vector
-}
\ No newline at end of file
+}
+
+TEST(CctpLibrary, TestRandomCustomFieldGeneration)
+{
+    srand(RANDOM_CUSTOM_FIELD_SEED);
+
+    for (int i = 0; i < 32; i++)
+    {
+        ASSERT_EQ(rand() % 256, RANDOM_CUSTOM_FIELD[i]);
+    }
+}
+
+TEST(CctpLibrary, TestGetLeadingZeros)
+{
+    ASSERT_EQ(8, getLeadingZeroBitsInByte(0));
+
+    for (uint8_t n = 1; n > 0; n++)
+    {
+        ASSERT_EQ(__builtin_clz(n) % (sizeof(unsigned int) * 8 - CHAR_BIT), getLeadingZeroBitsInByte(n));
+    }
+}
+
+TEST(CctpLibrary, TestCustomFieldsValidation)
+{
+    for (uint8_t i = 1; i < CHAR_BIT; i++)
+    {
+        for (uint8_t j = 1; j > 0; j++)
+        {
+            std::vector<unsigned char> rawBytes = { j };
+            FieldElementCertificateField certField = FieldElementCertificateField(rawBytes);
+            FieldElementCertificateFieldConfig config = FieldElementCertificateFieldConfig(i);
+            CFieldElement fe = certField.GetFieldElement(config);
+
+            if (j < 1 << i)
+            {
+                EXPECT_TRUE(fe.IsValid());
+            }
+            else
+            {
+                EXPECT_FALSE(fe.IsValid());
+            }
+        }
+    }
+}
+
+/**
+ * @brief Test the validation of a full (32 bytes) random custom field
+ * iteratively changing the last byte.
+ */
+TEST(CctpLibrary, TestFullCustomFieldValidation)
+{
+    std::vector<unsigned char> rawBytes(std::begin(RANDOM_CUSTOM_FIELD), std::end(RANDOM_CUSTOM_FIELD));
+
+    for (uint8_t i = 1; i < CHAR_BIT; i++)
+    {
+        for (uint8_t j = 1; j > 0; j++)
+        {
+            rawBytes.back() = j;
+            FieldElementCertificateField certField = FieldElementCertificateField(rawBytes);
+            FieldElementCertificateFieldConfig config = FieldElementCertificateFieldConfig(i + 31 * CHAR_BIT);
+            CFieldElement fe = certField.GetFieldElement(config);
+
+            // The Field Element is valid if it matches the configuration (j < 1 << i)
+            // and if doesn't exceed the modulus, thus the last two bits cannot be set (j < 1 << 6).
+            if (j < 1 << i && j < 1 << 6)
+            {
+                EXPECT_TRUE(fe.IsValid());
```

Comment:
This is correct, even if according to the implementation of GetFieldElement(config) the returned fe is always the null fe if the bits of cfg are not consistent (null fe is invalid anyway)